app/common/data/interfaces/collections.py

```diff
@@ -1121,32 +1121,40 @@ def update_question(
 
 @flush_and_rollback_on_exceptions
 def add_submission_event(
-    submission: Submission, key: SubmissionEventKey, user: User, form: Form | None = None
+    submission: Submission, *, event_type: SubmissionEventType, user: User, target_key: UUID | None = None
 ) -> Submission:
-    submission.events.append(SubmissionEvent(key=key, created_by=user, form=form))
+    submission.events.append(
+        SubmissionEvent(event_type=event_type, created_by=user, target_key=target_key if target_key else submission.id)
```

Comment:
nit:

```suggestion
        SubmissionEvent(event_type=event_type, created_by=user, target_key=target_key or submission.id)
```

tiny nit: if target_key is always a UUID (essentially a soft FK?), maybe related_entity_id?